src/main/java/bio/terra/app/controller/SnapshotsApiController.java

```diff
@@ -304,15 +303,15 @@ public ResponseEntity<SnapshotPreviewModel> lookupSnapshotPreviewById(
             .filter(filter));
   }
 
-  // --snapshot auth domains --
+  // --snapshot data access controls (auth domains, group access constraints) --
 
   @Override
   public ResponseEntity<AddAuthDomainResponseModel> addSnapshotAuthDomain(
       UUID id, List<String> userGroups) {
     AuthenticatedUserRequest userReq = getAuthenticatedInfo();
     verifySnapshotAuthorization(userReq, id.toString(), IamAction.UPDATE_AUTH_DOMAIN);
     AddAuthDomainResponseModel result =
-        snapshotService.addSnapshotAuthDomain(userReq, id, new ArrayList<>(userGroups));
```

Comment:
Have you tested out this change yet? I remember the flight failed to launch if I didn't cast `userGroups` to an ArrayList 🤔

Reply:
I tested this out on my dev environment with no issues!
```
PATCH https://jade-ok.datarepo-dev.broadinstitute.org/api/repository/v1/snapshots/0933cc3c-54c7-4b03-a16b-ee4cdeec8263/authDomain -d '["DataRepoAdmins"]'
{
  "authDomain": [
    "DataRepoAdmins"
  ]
}

GET https://jade-ok.datarepo-dev.broadinstitute.org/api/repository/v1/jobs/NwJUAvJzRMem81576IEl8A
{
  "id": "NwJUAvJzRMem81576IEl8A",
  "description": "Add data access control groups DataRepoAdmins to snapshot 0933cc3c-54c7-4b03-a16b-ee4cdeec8263",
  "job_status": "succeeded",
  "status_code": 200,
  "submitted": "2024-01-03T17:50:07.552874Z",
  "completed": "2024-01-03T17:50:09.070981Z",
  "class_name": "bio.terra.service.snapshot.flight.authDomain.SnapshotAddDataAccessControlsFlight"
}

GET https://jade-ok.datarepo-dev.broadinstitute.org/api/repository/v1/jobs/NwJUAvJzRMem81576IEl8A/result
{
  "authDomain": [
    "DataRepoAdmins"
  ]
}
```

Given this @samanehsan , are you okay if I merge?  Is there something else you'd like me to spot check?

Reply:
Thanks for spot checking! I think this is good to merge ✅